Mutagen.Bethesda.Analyzers.Engine/Engines/ContextualEngine.cs

```diff
@@ -58,16 +58,27 @@ public async Task Run()
                     listing.Mod,
                     modPath);
 
-                foreach (var driver in IsolatedModDrivers.Drivers)
+                await Task.WhenAll(IsolatedModDrivers.Drivers.Select(driver =>
                 {
-                    driver.Drive(isolatedParam);
-                }
+                    return _workDropoff.EnqueueAndWait(() =>
+                    {
+                        return driver.Drive(isolatedParam);
+                    }, cancel);
+                }));
             }
         }
 
-        foreach (var driver in ContextualModDrivers.Drivers)
+        var contextualParam = new ContextualDriverParams(
+            env.LinkCache,
+            env.LoadOrder,
+            ReportDropbox);
+
+        await Task.WhenAll(ContextualModDrivers.Drivers.Select(driver =>
```

Comment:
Yeah, we can probably improve this further.   Before i had things doing `Enqueue` without the `AndWait`, and in that mode, it didn't matter as the drivers above would quickly queue their work, and before it was done, it would get to this snippet and start off its work as well

However, that was causing problems in some areas (like environments getting disposed before their work was done).  I swapped back to `EnqueueAndWait`, as that's a lot more stable.  I was just trying to hyperoptimize since a new TaskCompletionSource needs to be made for each one.   Probably worth it for now, though